File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    answer = 0
    basic_babbling = ["aya", "ye", "woo", "ma"]
    for babble in babbling:
        for basic in basic_babbling:
            if basic*2 not in babble:
                if babble.__contains__(basic):
                    babble = babble.replace(basic, ' ')

        babble = babble.strip()
        if babble == '':
            answer += 1

    return answer

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<string> babbling) {
    int answer = 0;
    
    for(int i = 0; i < babbling.size(); i++) {
        int check = 1;
        string tmp = "";
        for(int j = 0; j < babbling[i].size(); j++){
            if (tmp != "aya" and babbling[i].substr(j,3)=="aya") j += 2, tmp = "aya";
            else if (tmp != "ye" and babbling[i].substr(j,2)=="ye") j += 1, tmp = "ye";
            else if (tmp != "woo" and babbling[i].substr(j,3)=="woo") j += 2, tmp = "woo";
            else if (tmp != "ma" and babbling[i].substr(j,2)=="ma") j += 1, tmp = "ma";
            else {
                check = 0;
                tmp = "";
                break;
            }
        }
        if (check == 1) answer++;
    }
    return answer;
}